clang/include/clang/Basic/AddressSpaces.h

```diff
@@ -87,6 +87,40 @@ inline bool isPtrSizeAddressSpace(LangAS AS) {
           AS == LangAS::ptr64);
 }
 
+inline LangAS getLangASasSYCLLangAS(LangAS AS) {
```

Comment:
The function name leaves a bit to be desired. How about: `getAddrSpaceAsSYCLAddrSpace()`?

Reply:
Yes it is a bit of a mouthful, however it matches the `asSYCLLangAS` naming from the `ParsedAttr` class making it much easier to search for all instances where something like this is done.

At the moment the switch between OpenCL and SYCL address spaces is done in two places, here as introduced by this patch and during parsing where it uses the `asSYCLLangAS` method.

Maybe we can go for just `asSYCLLangAS` here as well, what do you think?
 
I don't mind changing it to your suggestion either but I also think the better searchability is very valuable.

Reply:
I'd be fine with `asSYCLLangAS`, I was mostly trying to find ways to avoid `ASas` in the name because that's pretty hard to read. :-)